web-common/src/features/dashboards/time-controls/TimeControls.svelte

```diff
@@ -111,6 +111,9 @@
       timeGrain.grain,
       {}
     );
+
+    /** enable comparisons by default */
+    metricsExplorerStore.toggleComparison(metricViewName, true);
```

Comment:
I think this merits a bit more discussion. How these query libraries work is that "features" or "layouts" contain querying logic and pass things down into pure (or externally pure) components. The UI component that does the actual interaction should be "pure" or at least only maintain some simple UI state (e.g. this `intermediateSelection` variable).

This said, I'm not really thrilled with this section of the dashboard from a component composition POV and would prefer in the future we push this up as far as we can go, make the callbacks pure (i.e. no closing over state within a component), and have those callbacks in TS files so we can write tests against them (or just migrate any other logic directly into this `DashboardStore` element, which is effectively doing some Redux-y things.

As it is now, a lot of the dashboard code is pretty messy and inconsistent. Happy to satisfice on this PR and consider tactical refactors, especially now that we have more people.

Reply:
re: runtime awareness of the dashboard store, it's all client-side, and there's caching in the runtime at the data layer. This is because the actual `DashboardStore` itself is fairly transient state (at the moment). We encode it in a URL as a serialized protobuf. This is the only place we're currently saving it.

Reply:
The job of the dashboard store is to cache and save the UI state. It doesn't really need the runtime to do this

Reply:
I think theres two considerations to discuss here, and you've addressed one of them:
1. **Using query libraries that rely on components for indicating what data is needed** <- this is what you have addressed above. Many pros and cons to this. I can give you a decent argument for why it's preferable to have a centralized store that manages all query logic; the components interface with the store to get data or run an action but the store is what decides when/how to execute that stuff. MobX-State-Tree is a good example of this pattern https://mobx-state-tree.js.org/intro/welcome
2. **UI component regulating what the default values of the initial dashboard state are** - this is what I originally raised as an issue. Regardless of how you want to distribute querying logic across the app, its unclear to me why a UI component (<TimeControls />) would be responsible for enforcing constraints on the dashboard state. Defaults should be part of the dashboard state regardless of what UI components are on the screen